app/gui2/src/components/GraphEditor.vue

```diff
@@ -61,6 +61,21 @@ function keyboardBusy() {
   return document.activeElement != document.body
 }
 
+interface Interaction {
+  cancel?(): void
+  handleExprClick?(expr: ExprId): void
+  handleNodeClick?(node: ExprId): void
+}
+let currentInteraction: Interaction | null = null
+function setCurrentInteraction(interaction: Interaction | null) {
+  if (currentInteraction !== null) {
+    if (currentInteraction.cancel !== undefined) {
```

Comment:
prefer `!= null` instead of `!== null` and `!== undefined`.
(also, as mentioned elsewhere, prefer `undefined` over `null`. as a side effect that would make the `= null` initializer optional)